libp2p-peerstore/src/peer_info.rs

```diff
@@ -174,22 +159,13 @@ impl<'de> Deserialize<'de> for PeerInfo {
 
 		Ok(PeerInfo {
 			addrs: addrs,
-			public_key: interm.public_key,
 		})
 	}
 }
 
 impl PartialOrd for PeerInfo {
 	#[inline]
-	fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-		// See module-level comment.
-		match (&self.public_key, &other.public_key) {
-			(&Some(ref my_pub), &Some(ref other_pub)) => {
-				Some(my_pub.cmp(other_pub))
-			}
-			_ => {
-				None
-			}
-		}
+	fn partial_cmp(&self, _other: &Self) -> Option<Ordering> {
+		None
```

Comment:
is it a valid ordering?

Reply:
Technically it is valid to always return `None`.

The reason why we need to implement the `PartialOrd` trait is that the datastore library (a key-value storage) which we use allows performing queries where the results can be ordered. It was briefly debated whether we want to keep this feature (being able to order the results), but we decided to keep it despite the fact that it's IMO useless.

Since the struct that implements `PartialOrd` is internal and since we never use this ordering feature, I think it's ok to have this code.